client/src/contexts/FeedbackContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { getFromStorage, setInStorage } from 'API/localStorage';
+import { FeedbackData } from 'components/Data/FeedbackData';
+import React, { createContext, FC, useState, useMemo, useCallback, useEffect } from 'react';
+import { Feedback } from 'types';
+
+type FeedbackContextData = {
+  feedback: Feedback[];
+  setFeedback: (feedback: Feedback[]) => void;
+  addFeedback: (newFeedback: Feedback) => void;
+};
+
+const FeedbackContext = createContext<FeedbackContextData>({
+  feedback: [],
+  setFeedback: () => {
+    return null;
+  },
+  addFeedback: () => {
+    return null;
+  }
+});
+
+export const FeedbackProvider: FC<React.ReactNode> = ({ children }) => {
+  const [feedback, setFeedback] = useState<Feedback[]>([]);
+
+  const fetchFeedback = useCallback(() => {
+    const feedbackData = getFromStorage('feedback');
+    if (feedbackData) {
+      setFeedback(feedbackData);
+    } else {
+      setInStorage('feedback', FeedbackData);
+      setFeedback(FeedbackData);
+    }
+  }, [setFeedback]);
+
+  useEffect(() => {
+    fetchFeedback();
+  }, [fetchFeedback]);
+
+  const feedbackProvider = useMemo(() => {
+    const addFeedback = (newFeedback: Feedback) => {
+      setInStorage('feedback', [newFeedback, ...feedback]);
```

Comment:
cannot agree with point about customizing `setInStorage`, because it is defined as shared function and should be workable for different cases (for example, it can be used for another component with different logic of setting array in local storage)